### PR TITLE
fix: use multi-org settings layout even if not licensed

### PR DESCRIPTION
**Pull request by code-asher**
Originally, I had it using the multi-org layout if the multi-org experiment was enabled AND the feature was entitled.

However, we actually want to use the new layout even for single-org deployments, so I changed it; we now use the new layout ONLY when the multi-org experiment is enabled, even if the license is not entitled to the feature.  In the future, when the experiment is over, the multi-org layout will be the default and we can delete and permanently redirect the old views.

However, this means that you could see the create org page and not have the ability to use it, so I added our standard license badges to that page.  So if you go there and do not have a license, you now see the disabled badge.  

We did not have a badge for the premium license, which is what orgs requires, so I added it.  I am not 100% sure if the wording is right.

Then, the badges looked a bit janky with `PageHeader` so I moved the header used on all the other settings pages to the components directory and used it instead of `PageHeader`.  Then I used it on the rest of the org pages as well for consistency (create org, edit org, members, and groups).  It also brings some cohesiveness with the other settings pages, since they are all on the same sidebar and now these pages all use the same header style.  Doing this also got rid of some extra padding.

So in summary:

1. Should always use the new settings layout even if not licensed (as long as multi-org experiment is on).
2. Should see license info in the create org page.
3. Various org pages should have headers consistent with the other settings pages.

Stacked on: 
  - https://github.com/coder/coder/pull/14212

### CODE REVIEW

File: site/src/pages/ManagementSettingsPage/CreateOrganizationPageView.tsx [hunk 1]
```diff
@@ -54,9 +68,30 @@ export const CreateOrganizationPageView: FC<
...
+          <PopoverTrigger>
+            <span>
+              <PremiumBadge />
+            </span>
+          </PopoverTrigger>
+          <PopoverContent css={{ transform: "translateY(-28px)" }}>
+            <PopoverPaywall
+              message="Organizations"
+              description="Organizations allow you to run a Coder deployment with multiple platform teams, all with unique use cases, templates, and even underlying infrastructure."
+              // TODO: No documentation link yet.
```
Comment (code-asher) [inline site/src/pages/ManagementSettingsPage/CreateOrganizationPageView.tsx:88]: Will make a note

File: site/src/components/Paywall/PopoverPaywall.stories.tsx [hunk 2]
```diff
@@ -9,12 +9,19 @@ const meta: Meta<typeof PopoverPaywall> = {
...
       "Adds 3 mana of any single color of your choice to your mana pool, then is discarded. Tapping this artifact can be played as an interrupt.",
   },
 };
 
-export { Example as PopoverPaywall };
+export const Premium: Story = {
+  args: {
+    message: "Black Lotus",
+    description:
+      "Adds 3 mana of any single color of your choice to your mana pool, then is discarded. Tapping this artifact can be played as an interrupt.",
```
Comment (BrunoQuaresma) [inline site/src/components/Paywall/PopoverPaywall.stories.tsx:24]: LOL, someone is playing too much Magic 😆

**Review by BrunoQuaresma — APPROVED** [review]
It looks good to me. I just would make the header less bold like we do for the workspace proxies settings.

Screenshot 2024-08-08 at 10 41 21

File: site/src/components/Paywall/PopoverPaywall.stories.tsx (hunk 2 above)
Comment (Emyrk) [reply site/src/components/Paywall/PopoverPaywall.stories.tsx:24]: Choosing the MVP of cards

**Comment by Emyrk** [comment]
> However, this means that you could see the create org page and not have the ability to use it, so I added our standard license badges to that page. So if you go there and do not have a license, you now see the disabled badge.

Sweet!

**Comment by code-asher** [comment]
> It looks good to me. I just would make the header less bold like we do for the workspace proxies settings.

Right now the workspace proxies page is the only one that is not bold, so it looks like the odd one out.  Should we maybe change all of the pages to not be bold?

**Comment by BrunoQuaresma** [comment]
@code-asher I think so but I will leave this up to you

**Comment by code-asher** [comment]
> I think so but I will leave this up to you

Sounds good to me!  I will push up another PR soon for this.
